src/WithFileUploads.php

```diff
@@ -102,9 +102,17 @@ protected function dehydratePropertyFromWithFileUploads($name, $value)
     {
         if ($value instanceof TemporaryUploadedFile) {
             return $value->serializeForLivewireResponse();
-        } elseif (is_array($value) && isset(array_values($value)[0]) && array_values($value)[0] instanceof TemporaryUploadedFile) {
+        }
+
+        if (is_array($value) && isset(array_values($value)[0]) && array_values($value)[0] instanceof TemporaryUploadedFile && is_numeric(key($value))) {
             return array_values($value)[0]::serializeMultipleForLivewireResponse($value);
         }
+        
+        if (is_array($value)) {
+            foreach (array_keys($value) as $key) {
```

Comment:
why not use a regular `foreach ($value as $something)` instead of the key thing?

Reply:
we have to assign it back the dehydrated property of the value to its key and return the whole array. Maybe $key variable
`foreach ($value as $key => $item) {` will do